### PR TITLE
tests: enable unit tests for Python 3.9

### PR DESCRIPTION
**Pull request by tseaver**

### CODE REVIEW

File: synth.py [hunk 1]
```diff
@@ -71,7 +71,6 @@
 # ----------------------------------------------------------------------------
 templated_files = common.py_library(
     samples=False,  # set to True only if there are samples
-    unit_test_python_versions=["3.6", "3.7", "3.8"],
     system_test_python_versions=["3.7"],
```
Comment (crwilcox) [inline synth.py:74]: If we are templating this, your change is going to get overwritten. Should this line be updated not removed?

Comment (tseaver) [reply synth.py:74]: The templated change will Do the Right Thing(TM).

Comment (tseaver) [reply synth.py:74]: @crwilcox PTAL (if the value is not set, the defaults are correct).

**Comment by crwilcox** [comment]
closing as latest synths should cover this?

**Comment by tseaver** [comment]
```bash
$ git remote -v
origin	git@github.com:googleapis/python-firestore (fetch)
origin	git@github.com:googleapis/python-firestore (push)
$ git log -1
commit 403e1ba373c3e92044b926e9365f755c11101289 (HEAD -> master, origin/master, origin/HEAD)
Author: gcf-owl-bot[bot] <78513119+gcf-owl-bot[bot]@users.noreply.github.com>
Date:   Fri May 28 16:56:11 2021 +0000

    chore: new owl bot post processor docker image (#363)
    
    Post-Processor: gcr.io/repo-automation-bots/owlbot-python:latest@sha256:c66ba3c8d7bc8566f47df841f98cd0097b28fff0b1864c86f5817f4c8c3e8600
$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
$ nox --list
Sessions defined in /home/tseaver/projects/agendaless/Google/src/python-firestore/noxfile.py:

* lint -> Run linters.
* blacken -> Run black. Format code to uniform standard.
- pytype -> Run pytype
* lint_setup_py -> Verify that setup.py is valid (including RST check).
* unit-3.6 -> Run the unit test suite.
* unit-3.7 -> Run the unit test suite.
* unit-3.8 -> Run the unit test suite.
* system-3.7 -> Run the system test suite.
* cover -> Run the final coverage report.
* docs -> Build the docs for this library.
- docfx -> Build the docfx yaml files for this library.

sessions marked with * are selected, sessions marked with - are skipped.
```

**Pull request closed without merging**